### DISCUSSION TITLE
Avoid multiple commits to get pushed

### DISCUSSION THREAD

**Discussion by akkshitgupta** [EddieHubCommunity/LinkFree#8214] [Q&A] [answered]
I have a branch that contains the code for feature to implement. However, there were changes in the upstream during the feature development and when I updated my feature branch all the new commits came into my working tree as follows.

![image](https://github.com/EddieHubCommunity/LinkFree/assets/96991785/ecc6fdf3-0746-448f-930f-21a63cda39a1)

If you see in the above image, only 3 commits are mine out of 20. If I push the branch then all the commits will be pushed again.

How can I push only those commits that are mine?

**Comment by eddiejaoude** [comment]
Not sure how you got there without knowing the steps you took, but here are some tips...

- You should always branch from `main` and keep it up to date with the original repo
- Merge `main` into your branch frequently and do not force push

**Reply by akkshitgupta** [reply]
1. updated local `main` branch from upstream using 
```
git checkout main
git pull upstream main
```
2. merged local `main` branch into my working branch
```
git checkout <branch_name>
git merge main
``` 
after that I observed that there are some commits to be pushed and those were not mine.

**Reply by eddiejaoude** [reply, marked as answer]
That looks correct 👍 

The commits will be fine because they are in the `main`, when you do a diff or pull request, it will only show the difference

**Comment by akkshitgupta** [comment]
what about rebasing the head and dropping all the commits which are not required.

```
git rebase -i HEAD~n
```

interactive console opens and comment out those commits which are note required
